.travis.yml

```diff
@@ -1,11 +1,16 @@
 language: java
+env:
+  - VERIFY_USE_PUBLIC_BINARIES=true
 jdk:
   - oraclejdk8
   - oraclejdk9
   - openjdk8
 matrix:
   allow_failures:
   - jdk: oraclejdk9
-before_install:
-# this is a hack, so PR builds can be tested in travis
-  - perl -i -0pe 's/maven[\s\{]+[^\{\}]*\}/maven { url \"https:\/\/build.shibboleth.net\/nexus\/content\/groups\/public\" \n url \"https:\/\/repo1.maven.org\/maven2\" \n jcenter() }/gms' build.gradle
\ No newline at end of file
+services:
+  - docker
+before_script:
+  - docker build -f PublicDockerfile -t verify-event-emitter .
+script:
+  - docker run --env VERIFY_USE_PUBLIC_BINARIES verify-event-emitter /bin/sh -c "./gradlew clean build"
```

Comment:
`--env` allows Docker to fetch the initialised environment variable VERIFY_USE_PUBLIC_BINARIES for gradle since gradle is going to use the environment variable to choose whitelisted artifactory or bintray to download Java libraries for event emitter.